subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TaskBooleanOptionIntegrationTest.groovy

```diff
@@ -0,0 +1,219 @@
+        given:
+        file('buildSrc/src/main/java/SampleTask.java') << taskWithBooleanOptions()
+        buildFile << sampleTask()
+
+        when:
+        succeeds('help', '--task', 'sample')
+
+        then:
+        outputContains("""
+Detailed task information for sample
+
+Path
+     :sample
+
+Type
+     SampleTask (SampleTask)
+
+Options
+     --myBooleanObjectOption     Configures boolean option 'myBooleanObjectOption'
+
+     --myBooleanPrimitiveOption     Configures boolean option 'myBooleanPrimitiveOption'
+
+     --myBooleanPropertyOption     Configures Property<Boolean> option 'myBooleanPropertyOption'
+
+     --no-myBooleanObjectOption     Disables option --myBooleanObjectOption
+
+     --no-myBooleanPrimitiveOption     Disables option --myBooleanPrimitiveOption
+
+     --no-myBooleanPropertyOption     Disables option --myBooleanPropertyOption
+
+     --rerun     Causes the task to be re-run even if up-to-date.
+
+Description
+     -
+
+Group
+     -""")
+    }
+
+    static String sampleTask() {
+        """
+            task sample(type: SampleTask)
+        """
+    }
+
+    static String taskWithBooleanOptions() {
+        """
+            import org.gradle.api.DefaultTask;
+            import org.gradle.api.provider.Property;
+            import org.gradle.api.tasks.TaskAction;
+            import org.gradle.api.tasks.options.Option;
+            import org.gradle.api.tasks.options.OptionValues;
+
+            import java.util.Collections;
+            import java.util.List;
+            import java.util.Set;
+
+            public class SampleTask extends DefaultTask {
+                private boolean myBooleanPrimitiveOption;
+                private Boolean myBooleanObjectOption;
+                private Property<Boolean> myBooleanPropertyOption = getProject().getObjects().property(Boolean.class).convention((Boolean)null);;
+
+                public SampleTask() {}
+
+                @Option(option = "myBooleanPrimitiveOption", description = "Configures boolean option 'myBooleanPrimitiveOption'")
+                public void setMyBooleanPrimitiveOption(boolean myBooleanPrimitiveOption) {
+                    this.myBooleanPrimitiveOption = myBooleanPrimitiveOption;
+                }
+
+                @Option(option = "myBooleanObjectOption", description = "Configures boolean option 'myBooleanObjectOption'")
+                public void setMyBooleanObjectOption(Boolean myBooleanObjectOption) {
+                    this.myBooleanObjectOption = myBooleanObjectOption;
+                }
+
+                @Option(option = "myBooleanPropertyOption", description = "Configures Property<Boolean> option 'myBooleanPropertyOption'")
+                public void setMyBooleanPropertyOption(Boolean myBooleanPropertyOption) {
+                    this.myBooleanPropertyOption.set(myBooleanPropertyOption);
+                }
+
+                @OptionValues("myBooleanObjectOption")
```

Comment:
:question: What is this used for in this test?

Reply:
Good catch, a leftover from another test I used. Removed.